fili-core/src/main/java/com/yahoo/bard/webservice/application/healthchecks/LookupHealthCheck.java

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.application.healthchecks;
+
+import com.yahoo.bard.webservice.metadata.LookupMetadataLoadTask;
+
+import com.codahale.metrics.health.HealthCheck;
+
+import java.util.Set;
+
+import javax.inject.Singleton;
+
+/**
+ * Check load statuses of all Druid lookups.
+ */
+@Singleton
+public class LookupHealthCheck extends HealthCheck {
+    private final LookupMetadataLoadTask lookupMetadataLoadTask;
```

Comment:
Because this is potentially modified in another thread, we may need to make this volatile or make it a reference.